applications/matter_bridge/src/app_task.cpp

```diff
@@ -131,6 +136,29 @@ CHIP_ERROR AppTask::RestoreBridgedDevices()
 	return CHIP_NO_ERROR;
 }
 
+/* onoff plug part*/
+void AppTask::SmartplugOnOffEventHandler()
+{
+	SystemLayer().ScheduleLambda([] {
+		bool NewState = !Nrf::GetBoard().GetLED(Nrf::DeviceLeds::LED2).GetState();
```

Comment:
Isn't it a bit non-intuitive that we set the OnOff attribute based on the LED state (that was set based on the button pressed)? I suggest to extend the `SmartplugOnOffEventHandler` callback with the boolean input flag indicating the state and pass it directly in the `ButtonEventHandler`.

Reply:
It is, I must admit I did first in the light bulb :smile: However we need some variable to track the on-off state from the application level (not the one from Matter data model). We could add a new mState value in the AppTask and change its state in the button handler. But actually it would be redundant, because we already have such state, which is a state kept in the LED module. Maybe not perfect from the visual perspective, but how beautifully optimized it is!

Reply:
Ok, but it looks to we know the state of the button directly in the `ButtonEventHandler` (state, hasChanged), right?. So, can't we just forward this as a boolean flag to the `SmartplugOnOffEventHandler`? This would still be a state judged on the UI level, not data model.

Reply:
Yeah we know the button state, but we don't change LED state 1:1 based on that right? We use `state` value only to not trigger state changes on both signal edges. And we toggle the LED state on button press, so sometimes we will turn it off and sometimes on. In this case button state does not give us any information, we have to just negate the current LED state.